src/styles/index.scss

```diff
@@ -3 +1,53 @@
+@import './abstract/constants';
+@import './abstract/placeholders';
+@import './base/base';
+
+.calendar {
+  display: flex;
+  padding: $calendar-indent;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  @extend %calendar-width;
+
+  @each $day, $index in $days-of-week {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($day-size + $gap) * $index;
+    }
+  }
+
+  @for $length from $min-days through $max-days {
+    &--month-length-#{$length} {
+      @for $day from ($length) through $max-days {
+        .calendar__day:nth-child(#{$day + 1}) {
+          display: none;
+        }
+      }
+    }
+  }
```

Comment:
The logic in this loop is a bit confusing. It would be better to simplify it. For example, you can remove the inner loop and use the `:nth-child(n+#{day})` selector to hide all the days after the specified day.